### ISSUE TITLE
[java] PrematureDeclaration rule false-positive on variable declared to measure time

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#1208]
**Affects PMD Version:** 
6.4.0

**Rule:**
[PrematureDeclaration](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#prematuredeclaration)

**Description:**
PrematureDeclaration shouldn't trigger on classes which are specific to time measurement. E.g. it fails when stopwatch is used to measure time to invoke entire method (and I'm interested only in sunny-day scenario when exit points are not triggered).

**Code Sample demonstrating the issue:**

```java
Stopwatch stopwatch = Stopwatch.createStarted();
// some code with exit points
return stopwatch.elapsed(MILLISECONDS);
```
It fails on first line with `[PrematureDeclaration] Avoid declaring a variable if it is unreferenced before a possible exit point.`

**Comment by Andy-2639** [comment]
I would even extend this to axclude all calls with side effects
```
Object o = callWithSideEffects();
// code with exit points
doSomeThing(o);
```
from this rule. On the one hand, system calls (like reading time) cannot be moved without changing the behavior of the program. On the other hand, `callWithSideEffects` can do some work/change some state which is needed for the execution of `code with exit points`.

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
